### PR TITLE
[bug_5880] Added new tab

### PR DESCRIPTION
**Pull request by pwodzinski**
https://bugzilla.stxnext.pl/show_bug.cgi?id=5880

https://gist.github.com/pwodzinski/3185d54c0b4d0cb24283

### CODE REVIEW

File: src/presence_analyzer/static/js/totalhour.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            $.each(result, function(item) {
+                dropdown.append($("<option />").val(this.user_id).text(this.name));
+            });
+            dropdown.show();
+            loading.hide();
+        });
+        $('#user_id').change(function() {
+            var selected_user = $("#user_id").val(),
+                chart_div = $('#chart_div'),
+                userPhoto = $("#userPhoto").val('')
```
Comment (dpolcyn) [inline src/presence_analyzer/static/js/totalhour.js]: Semicolon

File: src/presence_analyzer/static/js/totalhour.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            if(selected_user) {
```
Comment (dpolcyn) [inline src/presence_analyzer/static/js/totalhour.js]: ```if (selected_user) {```

File: src/presence_analyzer/static/js/totalhour.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+                userPhoto.hide();
+                loading.show();
+                chart_div.hide();
+                $.getJSON("/api/v2/users", function(result) {
+                    $.each(result, function(item) {
+                        if (selected_user == this.user_id) {
+                            userPhoto = $("#userPhoto").attr("src", this.avatar)
```
Comment (dpolcyn) [inline src/presence_analyzer/static/js/totalhour.js]: Semicolon

File: src/presence_analyzer/static/js/totalhour.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+                        }
+                    });
+                    userPhoto.show();
+                });
+                $.getJSON("/api/v2/total_hour/"+selected_user, function(result) {
+                    var data = new google.visualization.arrayToDataTable(result);
+                    var options = {
+                        hAxis: {title: 'Weekday hours'}
+                    };
```
Comment (dpolcyn) [inline src/presence_analyzer/static/js/totalhour.js]: Use one var instead of two.

File: src/presence_analyzer/utils.py [hunk 5]
```diff
@@ -146,6 +146,22 @@ def parse_xml():
     return data
 
 
+def total_group_by_weekday(items):
+    """
+    Groups presence total users entries by weekday.
+    """
+    result = [[], [], [], [], [], [], []]  # one list for every day in week
```
Comment (dpolcyn) [inline src/presence_analyzer/utils.py]: ```result = [[] for i in range(7)]```

File: src/presence_analyzer/views.py [hunk 6]
```diff
@@ -120,6 +122,34 @@ def presence_weekday_view(user_id):
     return result
 
 
+@app.route('/api/v2/total_hour/<int:user_id>', methods=['GET'])
+@jsonify
+def presence_total_hour(user_id):
+    """
+    Returns user daily time and total daily time of all users
```
Comment (dpolcyn) [inline src/presence_analyzer/views.py]: Full stop on the end of line.

File: src/presence_analyzer/tests.py [hunk 7]
```diff
@@ -298,6 +322,26 @@ def test_group_by_weekday(self):
...
+        self.assertEqual(
+            weekdays,
+            [
+                24123,
+                46611,
+                49786,
+                69673,
+                6426,
+                0,
+                0
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: You can write this values in one line.

File: src/presence_analyzer/tests.py [hunk 8]
```diff
@@ -298,6 +322,26 @@ def test_group_by_weekday(self):
             ]
         )
 
+    def test_total_group_by_weekday(self):
+        """
+        Test grouping by weekday function for all users.
+        """
+        data = utils.get_data()
+        weekdays = utils.total_group_by_weekday(data)
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: ```data = utils.total_group_by_weekday(utils.get_data())```

File: src/presence_analyzer/tests.py [hunk 9]
```diff
@@ -50,6 +50,30 @@ def test_mainpage(self):
         self.assertEqual(resp.status_code, 302)
         assert resp.headers['Location'].endswith('/presence_weekday')
 
+    def test_total_hour(self):
+        """
+        Test api for total hours sorted by day
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: Full stop.

File: src/presence_analyzer/static/js/totalhour.js [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+                            userPhoto = $("#userPhoto").attr("src", this.avatar);
+                        }
+                    });
+                    userPhoto.show();
+                });
+                $.getJSON("/api/v2/total_hour/"+selected_user, function(result) {
+                    var data = new google.visualization.arrayToDataTable(result),
+                        options = {
+                        hAxis: {title: 'Weekday hours'}
+                    };
```
Comment (dpolcyn) [inline src/presence_analyzer/static/js/totalhour.js]: indentation

File: src/presence_analyzer/tests.py [hunk 11]
```diff
@@ -298,6 +322,17 @@ def test_group_by_weekday(self):
             ]
         )
 
+    def test_total_group_by_weekday(self):
+        """
+        Test grouping by weekday function for all users.
+        """
+        weekdays = utils.total_group_by_weekday(utils.get_data())
+        self.assertEqual(len(weekdays), 7)
```
Comment (mjelonek92) [inline src/presence_analyzer/tests.py]: You are already checking if the length is 7 line below this one by checking the whole content of weekdays.

File: src/presence_analyzer/views.py [hunk 12]
```diff
@@ -120,6 +122,34 @@ def presence_weekday_view(user_id):
...
+        log.debug('User %s not found!', user_id)
+        abort(404)
+
+    get_total_hours = total_group_by_weekday(data)
+
+    weekdays = group_by_weekday(data[user_id])
+    result = [
+        (
+            calendar.day_abbr[weekday],
+            float("%0.2f" % (float(sum(intervals))/3600)),
```
Comment (dpolcyn) [inline src/presence_analyzer/views.py]: ```float("%0.2f" % (float(sum(intervals)) / 3600)),```

**Pull request merged**